[FLINK-15614][docs] Consolidate Hadoop documentation (#10932)
apache/flink: docs/ops/deployment/hadoop.md

Another way to provide the Hadoop configuration is to have it on the class path of the Flink process, see more details below.

## Providing Hadoop classes

In order to use Hadoop features (e.g., YARN, HDFS) it is ncessary to provide Flink with the required Hadoop classes,
as these are not bundled by default.

This can be done in 2 ways:

> * Adding the Hadoop classpath to Flink

Author: Depends on the specific Hadoop distribution. The point is that user should first try the easy way, since building flink-shaded against custom/vendor hadoop versions is a) a pain and b) something we can't test for.

Another way to provide the Hadoop configuration is to have it on the class path of the Flink process, see more details below.

## Providing Hadoop classes

In order to use Hadoop features (e.g., YARN, HDFS) it is necessary to provide Flink with the required Hadoop classes,
as these are not bundled by default.

This can be done by 
1) Adding the Hadoop classpath to Flink
2) Putting the required jar files into /lib directory of the Flink distribution

> Option 1) requires very little work and integrates nicely with existing Hadoop setups, and should be the

Reviewer: ```suggestion
Option 1) requires very little work and integrates nicely with existing Hadoop setups. It should be the
```

preferred approach.

> However, Hadoop has a large dependency footprint, increasing the risk of dependency conflicts occurring.

Reviewer: ```suggestion
However, Hadoop has a large dependency footprint that increases the risk of dependency conflicts occurring.
```

Putting the Hadoop configuration in the same class path as the Hadoop libraries makes Flink pick up that configuration.

### Adding Hadoop to /lib

The Flink project releases Hadoop distributions for specific versions, that relocate or exclude several dependencies
to reduce the risk of dependency clashes.

> These can be found on the [downloads]({{ site.download_url }}).

Reviewer: ```suggestion
These can be found on the [downloads]({{ site.download_url }}) page in the optional components.
```

For these versions it is sufficient to download the corresponding `Pre-bundled Hadoop` component and placing it in
the `/lib` directory of the Flink distribution.

If the used Hadoop version is not listed on the download page (possibly due to being a Vendor-specific version),

> then it is necessary to build [flink-shaded](https://github.com/apache/flink-shaded) against this version.

Author: Doesn't make sense to refer to flink-shaded as a whole as a dependency; it's not, only parts of it are.

> You can find the source for this project in the [Additional Components]({{ site.download_url }}#additional-components) section of the download page.

Author: The website is the primary source for the projects source code and hence shouldn't be preceded by "also". It would imply that users _alternatively_ download things from the website.

<span class="label label-info">Note</span> If you want to build `flink-shaded` against a vendor specific Hadoop version, you first have to configure the
vendor-specific maven repository in your local maven setup as described [here](https://maven.apache.org/guides/mini/guide-multiple-repositories.html).

Run the following command to build and install `flink-shaded` against your desired Hadoop version (e.g., for version `2.6.5-custom`):

{% highlight bash %}
mvn clean install -Dhadoop.version=2.6.5-custom
{% endhighlight %}

> After this step is complete, place the `flink-shaded-hadoop-2-uber` jar in the `/lib` directory of the Flink distribution.

Author: not reasonably possible since the path contains the shaded-version which is independent of Flink.

If someone doesn't know which jar he should take he most likely didn't make it this far into the process anyway.
